<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MACSr • MACSr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MACSr">
<meta property="og:description" content="MACSr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/macs3-project/MACSr" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MACSr</h1>
            
      
      
      <div class="hidden name"><code>MACSr.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>With the improvement of sequencing techniques, chromatin
immunoprecipitation followed by high throughput sequencing (ChIP-Seq) is
getting popular to study genome-wide protein-DNA interactions. To
address the lack of powerful ChIP-Seq analysis method, we presented the
Model-based Analysis of ChIP-Seq (MACS), for identifying transcript
factor binding sites. MACS captures the influence of genome complexity
to evaluate the significance of enriched ChIP regions and MACS improves
the spatial resolution of binding sites through combining the
information of both sequencing tag position and orientation. MACS can be
easily used for ChIP-Seq data alone, or with a control sample with the
increase of specificity. Moreover, as a general peak-caller, MACS can
also be applied to any “DNA enrichment assays” if the question to be
asked is simply: where we can find significant reads coverage than the
random background.</p>
<p>This package is a wrapper of the MACS toolkit based on
<code>basilisk</code>.</p>
</div>
<div class="section level2">
<h2 id="load-the-package">Load the package<a class="anchor" aria-label="anchor" href="#load-the-package"></a>
</h2>
<p>The package is built on <a href="https://bioconductor.org/packages/release/bioc/html/basilisk.html" class="external-link">basilisk</a>.
The dependent python library <a href="https://github.com/macs3-project/MACS" class="external-link">macs3</a> will be installed
automatically inside its conda environment.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MACSr</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="section level3">
<h3 id="macs3-functions">MACS3 functions<a class="anchor" aria-label="anchor" href="#macs3-functions"></a>
</h3>
<p>There are 13 functions imported from MACS3. Details of each function
can be checked from its manual.</p>
<table class="table">
<colgroup>
<col width="19%">
<col width="80%">
</colgroup>
<thead><tr class="header">
<th>Functions</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>callpeak</code></td>
<td>Main MACS3 Function to call peaks from alignment results.</td>
</tr>
<tr class="even">
<td><code>bdgpeakcall</code></td>
<td>Call peaks from bedGraph output.</td>
</tr>
<tr class="odd">
<td><code>bdgbroadcall</code></td>
<td>Call broad peaks from bedGraph output.</td>
</tr>
<tr class="even">
<td><code>bdgcmp</code></td>
<td>Comparing two signal tracks in bedGraph format.</td>
</tr>
<tr class="odd">
<td><code>bdgopt</code></td>
<td>Operate the score column of bedGraph file.</td>
</tr>
<tr class="even">
<td><code>cmbreps</code></td>
<td>Combine BEDGraphs of scores from replicates.</td>
</tr>
<tr class="odd">
<td><code>bdgdiff</code></td>
<td>Differential peak detection based on paired four bedGraph
files.</td>
</tr>
<tr class="even">
<td><code>filterdup</code></td>
<td>Remove duplicate reads, then save in BED/BEDPE format.</td>
</tr>
<tr class="odd">
<td><code>predictd</code></td>
<td>Predict d or fragment size from alignment results.</td>
</tr>
<tr class="even">
<td><code>pileup</code></td>
<td>Pileup aligned reads (single-end) or fragments (paired-end)</td>
</tr>
<tr class="odd">
<td><code>randsample</code></td>
<td>Randomly choose a number/percentage of total reads.</td>
</tr>
<tr class="even">
<td><code>refinepeak</code></td>
<td>Take raw reads alignment, refine peak summits.</td>
</tr>
<tr class="odd">
<td><code>callvar</code></td>
<td>Call variants in given peak regions from the alignment BAM
files.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="function-callpeak">Function <code>callpeak</code><a class="anchor" aria-label="anchor" href="#function-callpeak"></a>
</h3>
<p>We have uploaded multipe test datasets from MACS to a data package
<code>MACSdata</code> in the <code>ExperimentHub</code>. For example,
Here we download a pair of single-end bed files to run the
<code>callpeak</code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="st">"MACSdata"</span><span class="op">)</span></span>
<span><span class="va">CHIP</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4558"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; see ?MACSdata and browseVignettes('MACSdata') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">CTRL</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4563"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; see ?MACSdata and browseVignettes('MACSdata') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span></code></pre></div>
<p>Here is an example to call narrow and broad peaks on the SE bed
files.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cp1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>, store_bdg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                name <span class="op">=</span> <span class="st">"run_callpeak_narrow0"</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                cutoff_analysis <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [635 MB] </span></span>
<span><span class="co">#&gt; # Command line: </span></span>
<span><span class="co">#&gt; # ARGUMENTS LIST:</span></span>
<span><span class="co">#&gt; # name = run_callpeak_narrow0</span></span>
<span><span class="co">#&gt; # format = AUTO</span></span>
<span><span class="co">#&gt; # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601']</span></span>
<span><span class="co">#&gt; # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606']</span></span>
<span><span class="co">#&gt; # effective genome size = 5.20e+07</span></span>
<span><span class="co">#&gt; # band width = 300</span></span>
<span><span class="co">#&gt; # model fold = [5.0, 50.0]</span></span>
<span><span class="co">#&gt; # qvalue cutoff = 5.00e-02</span></span>
<span><span class="co">#&gt; # The maximum gap between significant sites is assigned as the read length/tag size.</span></span>
<span><span class="co">#&gt; # The minimum length of peaks is assigned as the predicted fragment length "d".</span></span>
<span><span class="co">#&gt; # Larger dataset will be scaled towards smaller dataset.</span></span>
<span><span class="co">#&gt; # Range for calculating regional lambda is: 1000 bps and 10000 bps</span></span>
<span><span class="co">#&gt; # Broad region calling is off</span></span>
<span><span class="co">#&gt; # Additional cutoff on fold-enrichment is: 0.10</span></span>
<span><span class="co">#&gt; # Paired-End mode is off</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [635 MB] #1 read tag files... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [635 MB] #1 read treatment tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [638 MB] Detected format is: BED </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [638 MB] * Input file is gzipped. </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [642 MB] #1.2 read input tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [642 MB] Detected format is: BED </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [642 MB] * Input file is gzipped. </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 tag size is determined as 101 bps </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 tag size = 101.0 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  total tags in treatment: 49622 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 user defined the maximum tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  tags after filtering in treatment: 48047 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  Redundant rate of treatment: 0.03 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  total tags in control: 50837 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 user defined the maximum tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  tags after filtering in control: 50783 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  Redundant rate of control: 0.00 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 finished! </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 Build Peak Model... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 looking for paired plus/minus strand peaks... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 Total number of paired peaks: 469 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 Model building with cross-correlation: Done </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 finished! </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 predicted fragment length is 228 bps </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 alternative fragment length(s) may be 228 bps </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2.2 Generate R script for model : /tmp/Rtmp4jg3ab/run_callpeak_narrow0_model.r </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #3 Call peaks... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #3 Pre-compute pvalue-qvalue table... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [643 MB] #3 Cutoff vs peaks called will be analyzed! </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/tmp/Rtmp4jg3ab/run_callpeak_narrow0_cutoff_analysis.txt' </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3 In the peak calling step, the following will be performed simultaneously: </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3   Write bedGraph files for treatment pileup (after scaling if necessary)... run_callpeak_narrow0_treat_pileup.bdg </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3   Write bedGraph files for control lambda (after scaling if necessary)... run_callpeak_narrow0_control_lambda.bdg </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3   Pileup will be based on sequencing depth in treatment. </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3 Call peaks for each chromosome... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #4 Write output xls file... /tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.xls </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #4 Write peak in narrowPeak format file... /tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.narrowPeak </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #4 Write summits bed file... /tmp/Rtmp4jg3ab/run_callpeak_narrow0_summits.bed </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] Done!</span></span>
<span><span class="va">cp2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>, store_bdg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                name <span class="op">=</span> <span class="st">"run_callpeak_broad"</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                broad <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] </span></span>
<span><span class="co">#&gt; # Command line: </span></span>
<span><span class="co">#&gt; # ARGUMENTS LIST:</span></span>
<span><span class="co">#&gt; # name = run_callpeak_broad</span></span>
<span><span class="co">#&gt; # format = AUTO</span></span>
<span><span class="co">#&gt; # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601']</span></span>
<span><span class="co">#&gt; # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606']</span></span>
<span><span class="co">#&gt; # effective genome size = 5.20e+07</span></span>
<span><span class="co">#&gt; # band width = 300</span></span>
<span><span class="co">#&gt; # model fold = [5.0, 50.0]</span></span>
<span><span class="co">#&gt; # qvalue cutoff for narrow/strong regions = 5.00e-02</span></span>
<span><span class="co">#&gt; # qvalue cutoff for broad/weak regions = 1.00e-01</span></span>
<span><span class="co">#&gt; # The maximum gap between significant sites is assigned as the read length/tag size.</span></span>
<span><span class="co">#&gt; # The minimum length of peaks is assigned as the predicted fragment length "d".</span></span>
<span><span class="co">#&gt; # Larger dataset will be scaled towards smaller dataset.</span></span>
<span><span class="co">#&gt; # Range for calculating regional lambda is: 1000 bps and 10000 bps</span></span>
<span><span class="co">#&gt; # Broad region calling is on</span></span>
<span><span class="co">#&gt; # Additional cutoff on fold-enrichment is: 0.10</span></span>
<span><span class="co">#&gt; # Paired-End mode is off</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 read tag files... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 read treatment tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] Detected format is: BED </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] * Input file is gzipped. </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1.2 read input tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] Detected format is: BED </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] * Input file is gzipped. </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 tag size is determined as 101 bps </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 tag size = 101.0 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1  total tags in treatment: 49622 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 user defined the maximum tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1  tags after filtering in treatment: 48047 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1  Redundant rate of treatment: 0.03 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1  total tags in control: 50837 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 user defined the maximum tags... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1  tags after filtering in control: 50783 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1  Redundant rate of control: 0.00 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #1 finished! </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 Build Peak Model... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 looking for paired plus/minus strand peaks... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 Total number of paired peaks: 469 </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 Model building with cross-correlation: Done </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 finished! </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 predicted fragment length is 228 bps </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2 alternative fragment length(s) may be 228 bps </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #2.2 Generate R script for model : /tmp/Rtmp4jg3ab/run_callpeak_broad_model.r </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #3 Call peaks... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [664 MB] #3 Pre-compute pvalue-qvalue table... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #3 In the peak calling step, the following will be performed simultaneously: </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #3   Write bedGraph files for treatment pileup (after scaling if necessary)... run_callpeak_broad_treat_pileup.bdg </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #3   Write bedGraph files for control lambda (after scaling if necessary)... run_callpeak_broad_control_lambda.bdg </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #3 Call peaks for each chromosome... </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #4 Write output xls file... /tmp/Rtmp4jg3ab/run_callpeak_broad_peaks.xls </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #4 Write broad peak in broadPeak format file... /tmp/Rtmp4jg3ab/run_callpeak_broad_peaks.broadPeak </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] #4 Write broad peak in bed12/gappedPeak format file... /tmp/Rtmp4jg3ab/run_callpeak_broad_peaks.gappedPeak </span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:34: [664 MB] Done!</span></span></code></pre></div>
<p>Here are the outputs.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cp1</span></span>
<span><span class="co">#&gt; macsList class</span></span>
<span><span class="co">#&gt; $outputs:</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_control_lambda.bdg</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_cutoff_analysis.txt</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_model.r</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.narrowPeak</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.xls</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_summits.bed</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_narrow0_treat_pileup.bdg </span></span>
<span><span class="co">#&gt; $arguments: tfile, cfile, gsize, outdir, name, store_bdg, cutoff_analysis </span></span>
<span><span class="co">#&gt; $log:</span></span>
<span><span class="co">#&gt;  INFO  @ 17 Nov 2023 12:43:33: [635 MB] </span></span>
<span><span class="co">#&gt;  # Command line: </span></span>
<span><span class="co">#&gt;  # ARGUMENTS LIST:</span></span>
<span><span class="co">#&gt;  # name = run_callpeak_narrow0</span></span>
<span><span class="co">#&gt;  # format = AUTO</span></span>
<span><span class="co">#&gt; ...</span></span>
<span><span class="va">cp2</span></span>
<span><span class="co">#&gt; macsList class</span></span>
<span><span class="co">#&gt; $outputs:</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_broad_control_lambda.bdg</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_broad_model.r</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_broad_peaks.broadPeak</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_broad_peaks.gappedPeak</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_broad_peaks.xls</span></span>
<span><span class="co">#&gt;  /tmp/Rtmp4jg3ab/run_callpeak_broad_treat_pileup.bdg </span></span>
<span><span class="co">#&gt; $arguments: tfile, cfile, gsize, outdir, name, store_bdg, broad </span></span>
<span><span class="co">#&gt; $log:</span></span>
<span><span class="co">#&gt;  INFO  @ 17 Nov 2023 12:43:33: [664 MB] </span></span>
<span><span class="co">#&gt;  # Command line: </span></span>
<span><span class="co">#&gt;  # ARGUMENTS LIST:</span></span>
<span><span class="co">#&gt;  # name = run_callpeak_broad</span></span>
<span><span class="co">#&gt;  # format = AUTO</span></span>
<span><span class="co">#&gt; ...</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="the-macslist-class">The <code>macsList</code> class<a class="anchor" aria-label="anchor" href="#the-macslist-class"></a>
</h3>
<p>The <code>macsList</code> is designed to contain everything of an
execution, including function, inputs, outputs and logs, for the purpose
of reproducibility.</p>
<p>For example, we can the function and input arguments.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cp1</span><span class="op">$</span><span class="va">arguments</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; callpeak</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $tfile</span></span>
<span><span class="co">#&gt; CHIP</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $cfile</span></span>
<span><span class="co">#&gt; CTRL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $gsize</span></span>
<span><span class="co">#&gt; [1] 5.2e+07</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $outdir</span></span>
<span><span class="co">#&gt; tempdir()</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $name</span></span>
<span><span class="co">#&gt; [1] "run_callpeak_narrow0"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $store_bdg</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $cutoff_analysis</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>The files of all the outputs are collected.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cp1</span><span class="op">$</span><span class="va">outputs</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_control_lambda.bdg" </span></span>
<span><span class="co">#&gt; [2] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_cutoff_analysis.txt"</span></span>
<span><span class="co">#&gt; [3] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_model.r"            </span></span>
<span><span class="co">#&gt; [4] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.narrowPeak"   </span></span>
<span><span class="co">#&gt; [5] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.xls"          </span></span>
<span><span class="co">#&gt; [6] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_summits.bed"        </span></span>
<span><span class="co">#&gt; [7] "/tmp/Rtmp4jg3ab/run_callpeak_narrow0_treat_pileup.bdg"</span></span></code></pre></div>
<p>The <code>log</code> is especially important for <code>MACS</code> to
check. Detailed information was given in the log when running.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">cp1</span><span class="op">$</span><span class="va">log</span><span class="op">)</span></span>
<span><span class="co">#&gt; INFO  @ 17 Nov 2023 12:43:33: [635 MB]  # Command line:  # ARGUMENTS LIST: # name = run_callpeak_narrow0 # format = AUTO # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601'] # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606'] # effective genome size = 5.20e+07 # band width = 300 # model fold = [5.0, 50.0] # qvalue cutoff = 5.00e-02 # The maximum gap between significant sites is assigned as the read length/tag size. # The minimum length of peaks is assigned as the predicted fragment length "d". # Larger dataset will be scaled towards smaller dataset. # Range for calculating regional lambda is: 1000 bps and 10000 bps # Broad region calling is off # Additional cutoff on fold-enrichment is: 0.10 # Paired-End mode is off   INFO  @ 17 Nov 2023 12:43:33: [635 MB] #1 read tag files...  INFO  @ 17 Nov 2023 12:43:33: [635 MB] #1 read treatment tags...  INFO  @ 17 Nov 2023 12:43:33: [638 MB] Detected format is: BED  INFO  @ 17 Nov 2023 12:43:33: [638 MB] * Input file is gzipped.  INFO  @ 17 Nov 2023 12:43:33: [642 MB] #1.2 read input tags...  INFO  @ 17 Nov 2023 12:43:33: [642 MB] Detected format is: BED  INFO  @ 17 Nov 2023 12:43:33: [642 MB] * Input file is gzipped.  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 tag size is determined as 101 bps  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 tag size = 101.0  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  total tags in treatment: 49622  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 user defined the maximum tags...  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s)  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  tags after filtering in treatment: 48047  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  Redundant rate of treatment: 0.03  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  total tags in control: 50837  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 user defined the maximum tags...  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s)  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  tags after filtering in control: 50783  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1  Redundant rate of control: 0.00  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #1 finished!  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 Build Peak Model...  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 looking for paired plus/minus strand peaks...  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 Total number of paired peaks: 469  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 Model building with cross-correlation: Done  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 finished!  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 predicted fragment length is 228 bps  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2 alternative fragment length(s) may be 228 bps  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #2.2 Generate R script for model : /tmp/Rtmp4jg3ab/run_callpeak_narrow0_model.r  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #3 Call peaks...  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #3 Pre-compute pvalue-qvalue table...  INFO  @ 17 Nov 2023 12:43:33: [643 MB] #3 Cutoff vs peaks called will be analyzed!  INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/tmp/Rtmp4jg3ab/run_callpeak_narrow0_cutoff_analysis.txt'  INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3 In the peak calling step, the following will be performed simultaneously:  INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3   Write bedGraph files for treatment pileup (after scaling if necessary)... run_callpeak_narrow0_treat_pileup.bdg  INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3   Write bedGraph files for control lambda (after scaling if necessary)... run_callpeak_narrow0_control_lambda.bdg  INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3   Pileup will be based on sequencing depth in treatment.  INFO  @ 17 Nov 2023 12:43:33: [663 MB] #3 Call peaks for each chromosome...  INFO  @ 17 Nov 2023 12:43:33: [664 MB] #4 Write output xls file... /tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.xls  INFO  @ 17 Nov 2023 12:43:33: [664 MB] #4 Write peak in narrowPeak format file... /tmp/Rtmp4jg3ab/run_callpeak_narrow0_peaks.narrowPeak  INFO  @ 17 Nov 2023 12:43:33: [664 MB] #4 Write summits bed file... /tmp/Rtmp4jg3ab/run_callpeak_narrow0_summits.bed  INFO  @ 17 Nov 2023 12:43:33: [664 MB] Done!</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="resources">Resources<a class="anchor" aria-label="anchor" href="#resources"></a>
</h2>
<p>More details about <code>MACS3</code> can be found: <a href="https://macs3-project.github.io/MACS/" class="external-link uri">https://macs3-project.github.io/MACS/</a>.</p>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: /home/qhu/miniconda3/envs/rcwl/lib/libopenblasp-r0.3.23.so;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] MACSdata_1.8.0   MACSr_1.11.2     BiocStyle_2.28.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0              dplyr_1.1.2                  </span></span>
<span><span class="co">#&gt;  [3] blob_1.2.4                    bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;  [5] filelock_1.0.2                Biostrings_2.68.1            </span></span>
<span><span class="co">#&gt;  [7] fastmap_1.1.1                 RCurl_1.98-1.13              </span></span>
<span><span class="co">#&gt;  [9] BiocFileCache_2.8.0           promises_1.2.0.1             </span></span>
<span><span class="co">#&gt; [11] digest_0.6.33                 mime_0.12                    </span></span>
<span><span class="co">#&gt; [13] lifecycle_1.0.3               ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [15] KEGGREST_1.40.1               interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">#&gt; [17] RSQLite_2.3.1                 magrittr_2.0.3               </span></span>
<span><span class="co">#&gt; [19] compiler_4.3.1                rlang_1.1.1                  </span></span>
<span><span class="co">#&gt; [21] sass_0.4.7                    tools_4.3.1                  </span></span>
<span><span class="co">#&gt; [23] utf8_1.2.3                    yaml_2.3.7                   </span></span>
<span><span class="co">#&gt; [25] knitr_1.43                    bit_4.0.5                    </span></span>
<span><span class="co">#&gt; [27] curl_5.0.1                    here_1.0.1                   </span></span>
<span><span class="co">#&gt; [29] reticulate_1.30               withr_2.5.0                  </span></span>
<span><span class="co">#&gt; [31] purrr_1.0.1                   BiocGenerics_0.46.0          </span></span>
<span><span class="co">#&gt; [33] desc_1.4.2                    grid_4.3.1                   </span></span>
<span><span class="co">#&gt; [35] stats4_4.3.1                  fansi_1.0.4                  </span></span>
<span><span class="co">#&gt; [37] ExperimentHub_2.10.0          xtable_1.8-4                 </span></span>
<span><span class="co">#&gt; [39] cli_3.6.1                     rmarkdown_2.25               </span></span>
<span><span class="co">#&gt; [41] crayon_1.5.2                  ragg_1.2.5                   </span></span>
<span><span class="co">#&gt; [43] generics_0.1.3                httr_1.4.6                   </span></span>
<span><span class="co">#&gt; [45] DBI_1.1.3                     cachem_1.0.8                 </span></span>
<span><span class="co">#&gt; [47] stringr_1.5.0                 zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt; [49] parallel_4.3.1                AnnotationDbi_1.62.2         </span></span>
<span><span class="co">#&gt; [51] BiocManager_1.30.21.1         XVector_0.40.0               </span></span>
<span><span class="co">#&gt; [53] basilisk_1.12.1               vctrs_0.6.3                  </span></span>
<span><span class="co">#&gt; [55] Matrix_1.6-0                  jsonlite_1.8.7               </span></span>
<span><span class="co">#&gt; [57] dir.expiry_1.8.0              bookdown_0.36                </span></span>
<span><span class="co">#&gt; [59] IRanges_2.34.1                S4Vectors_0.38.1             </span></span>
<span><span class="co">#&gt; [61] bit64_4.0.5                   systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [63] jquerylib_0.1.4               glue_1.6.2                   </span></span>
<span><span class="co">#&gt; [65] pkgdown_2.0.7                 stringi_1.7.12               </span></span>
<span><span class="co">#&gt; [67] BiocVersion_3.17.1            later_1.3.1                  </span></span>
<span><span class="co">#&gt; [69] GenomeInfoDb_1.36.4           tibble_3.2.1                 </span></span>
<span><span class="co">#&gt; [71] pillar_1.9.0                  basilisk.utils_1.12.1        </span></span>
<span><span class="co">#&gt; [73] rappdirs_0.3.3                htmltools_0.5.5              </span></span>
<span><span class="co">#&gt; [75] GenomeInfoDbData_1.2.10       R6_2.5.1                     </span></span>
<span><span class="co">#&gt; [77] dbplyr_2.3.3                  textshaping_0.3.6            </span></span>
<span><span class="co">#&gt; [79] rprojroot_2.0.3               evaluate_0.21                </span></span>
<span><span class="co">#&gt; [81] shiny_1.7.4.1                 lattice_0.21-8               </span></span>
<span><span class="co">#&gt; [83] Biobase_2.60.0                AnnotationHub_3.10.0         </span></span>
<span><span class="co">#&gt; [85] png_0.1-8                     memoise_2.0.1                </span></span>
<span><span class="co">#&gt; [87] httpuv_1.6.11                 bslib_0.5.0                  </span></span>
<span><span class="co">#&gt; [89] Rcpp_1.0.11                   xfun_0.39                    </span></span>
<span><span class="co">#&gt; [91] fs_1.6.3                      pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
