<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MACSr • MACSr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MACSr">
<meta property="og:description" content="MACSr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/macs3-project/MACSr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MACSr_files/header-attrs-2.5/header-attrs.js"></script><link href="MACSr_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="MACSr_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MACSr</h1>
            
      
      
      <div class="hidden name"><code>MACSr.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>With the improvement of sequencing techniques, chromatin immunoprecipitation followed by high throughput sequencing (ChIP-Seq) is getting popular to study genome-wide protein-DNA interactions. To address the lack of powerful ChIP-Seq analysis method, we presented the Model-based Analysis of ChIP-Seq (MACS), for identifying transcript factor binding sites. MACS captures the influence of genome complexity to evaluate the significance of enriched ChIP regions and MACS improves the spatial resolution of binding sites through combining the information of both sequencing tag position and orientation. MACS can be easily used for ChIP-Seq data alone, or with a control sample with the increase of specificity. Moreover, as a general peak-caller, MACS can also be applied to any “DNA enrichment assays” if the question to be asked is simply: where we can find significant reads coverage than the random background.</p>
<p>This package is a wrapper of the MACS toolkit based on <code>basilisk</code>.</p>
</div>
<div id="load-the-package" class="section level1">
<h1 class="hasAnchor">
<a href="#load-the-package" class="anchor"></a>Load the package</h1>
<p>The package is built on <a href="https://bioconductor.org/packages/release/bioc/html/basilisk.html">basilisk</a>. The dependent python library <a href="https://github.com/macs3-project/MACS">macs3</a> will be installed automatically inside its conda environment.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MACSr</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<div id="macs3-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#macs3-functions" class="anchor"></a>MACS3 functions</h2>
<p>There are 13 functions imported from MACS3. Details of each function can be checked from its manual.</p>
<table class="table">
<thead><tr class="header">
<th>Functions</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>callpeak</code></td>
<td>Main MACS3 Function to call peaks from alignment results.</td>
</tr>
<tr class="even">
<td><code>bdgpeakcall</code></td>
<td>Call peaks from bedGraph output.</td>
</tr>
<tr class="odd">
<td><code>bdgbroadcall</code></td>
<td>Call broad peaks from bedGraph output.</td>
</tr>
<tr class="even">
<td><code>bdgcmp</code></td>
<td>Comparing two signal tracks in bedGraph format.</td>
</tr>
<tr class="odd">
<td><code>bdgopt</code></td>
<td>Operate the score column of bedGraph file.</td>
</tr>
<tr class="even">
<td><code>cmbreps</code></td>
<td>Combine BEDGraphs of scores from replicates.</td>
</tr>
<tr class="odd">
<td><code>bdgdiff</code></td>
<td>Differential peak detection based on paired four bedGraph files.</td>
</tr>
<tr class="even">
<td><code>filterdup</code></td>
<td>Remove duplicate reads, then save in BED/BEDPE format.</td>
</tr>
<tr class="odd">
<td><code>predictd</code></td>
<td>Predict d or fragment size from alignment results.</td>
</tr>
<tr class="even">
<td><code>pileup</code></td>
<td>Pileup aligned reads (single-end) or fragments (paired-end)</td>
</tr>
<tr class="odd">
<td><code>randsample</code></td>
<td>Randomly choose a number/percentage of total reads.</td>
</tr>
<tr class="even">
<td><code>refinepeak</code></td>
<td>Take raw reads alignment, refine peak summits.</td>
</tr>
<tr class="odd">
<td><code>callvar</code></td>
<td>Call variants in given peak regions from the alignment BAM files.</td>
</tr>
</tbody>
</table>
</div>
<div id="function-callpeak" class="section level2">
<h2 class="hasAnchor">
<a href="#function-callpeak" class="anchor"></a>Function <code>callpeak</code>
</h2>
<p>We have uploaded multipe test datasets from MACS to a data package <code>MACSdata</code> in the <code>ExperimentHub</code>. For example, Here we download a pair of single-end bed files to run the <code>callpeak</code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; snapshotDate(): 2020-12-09</span>
<span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="st">"MACSdata"</span><span class="op">)</span>
<span class="va">CHIP</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4558"</span><span class="op">]</span><span class="op">]</span>
<span class="co">#&gt; see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="co">#&gt; loading from cache</span>
<span class="va">CTRL</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4563"</span><span class="op">]</span><span class="op">]</span>
<span class="co">#&gt; see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="co">#&gt; loading from cache</span></code></pre></div>
<p>Here is an example to call narrow and broad peaks on the SE bed files.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cp1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>, store_bdg <span class="op">=</span> <span class="cn">TRUE</span>,
                name <span class="op">=</span> <span class="st">"run_callpeak_narrow0"</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>,
                cutoff_analysis <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: </span>
<span class="co">#&gt; # Command line: </span>
<span class="co">#&gt; # ARGUMENTS LIST:</span>
<span class="co">#&gt; # name = run_callpeak_narrow0</span>
<span class="co">#&gt; # format = AUTO</span>
<span class="co">#&gt; # ChIP-seq file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c68423d6b_4601']</span>
<span class="co">#&gt; # control file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c54ce7146_4606']</span>
<span class="co">#&gt; # effective genome size = 5.20e+07</span>
<span class="co">#&gt; # band width = 300</span>
<span class="co">#&gt; # model fold = [5.0, 50.0]</span>
<span class="co">#&gt; # qvalue cutoff = 5.00e-02</span>
<span class="co">#&gt; # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="co">#&gt; # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="co">#&gt; # Larger dataset will be scaled towards smaller dataset.</span>
<span class="co">#&gt; # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="co">#&gt; # Broad region calling is off</span>
<span class="co">#&gt; # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="co">#&gt; # Paired-End mode is off</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 read tag files... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 read treatment tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: Detected format is: BED </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: * Input file is gzipped. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1.2 read input tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: Detected format is: BED </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: * Input file is gzipped. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 tag size is determined as 101 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 tag size = 101.0 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  total tags in treatment: 49622 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 user defined the maximum tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  tags after filtering in treatment: 48047 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  Redundant rate of treatment: 0.03 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  total tags in control: 50837 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 user defined the maximum tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  tags after filtering in control: 50783 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  Redundant rate of control: 0.00 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 finished! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 Build Peak Model... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 looking for paired plus/minus strand peaks... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 Total number of paired peaks: 1036 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 Model building with cross-correlation: Done </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 finished! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 predicted fragment length is 230 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 alternative fragment length(s) may be 230,238 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2.2 Generate R script for model : /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_model.r </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #3 Call peaks... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #3 Pre-compute pvalue-qvalue table... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #3 Cutoff vs peaks called will be analyzed! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_cutoff_analysis.txt' </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 In the peak calling step, the following will be performed simultaneously: </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... run_callpeak_narrow0_treat_pileup.bdg </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Write bedGraph files for control lambda (after scaling if necessary)... run_callpeak_narrow0_control_lambda.bdg </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Pileup will be based on sequencing depth in treatment. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Call peaks for each chromosome... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #4 Write output xls file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_peaks.xls </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #4 Write peak in narrowPeak format file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_peaks.narrowPeak </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #4 Write summits bed file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_summits.bed </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: Done!</span>
<span class="va">cp2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>, store_bdg <span class="op">=</span> <span class="cn">TRUE</span>,
                name <span class="op">=</span> <span class="st">"run_callpeak_broad"</span>, outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>,
                broad <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: </span>
<span class="co">#&gt; # Command line: </span>
<span class="co">#&gt; # ARGUMENTS LIST:</span>
<span class="co">#&gt; # name = run_callpeak_broad</span>
<span class="co">#&gt; # format = AUTO</span>
<span class="co">#&gt; # ChIP-seq file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c68423d6b_4601']</span>
<span class="co">#&gt; # control file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c54ce7146_4606']</span>
<span class="co">#&gt; # effective genome size = 5.20e+07</span>
<span class="co">#&gt; # band width = 300</span>
<span class="co">#&gt; # model fold = [5.0, 50.0]</span>
<span class="co">#&gt; # qvalue cutoff for narrow/strong regions = 5.00e-02</span>
<span class="co">#&gt; # qvalue cutoff for broad/weak regions = 1.00e-01</span>
<span class="co">#&gt; # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="co">#&gt; # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="co">#&gt; # Larger dataset will be scaled towards smaller dataset.</span>
<span class="co">#&gt; # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="co">#&gt; # Broad region calling is on</span>
<span class="co">#&gt; # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="co">#&gt; # Paired-End mode is off</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 read tag files... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 read treatment tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: Detected format is: BED </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: * Input file is gzipped. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1.2 read input tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: Detected format is: BED </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: * Input file is gzipped. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 tag size is determined as 101 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 tag size = 101.0 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1  total tags in treatment: 49622 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 user defined the maximum tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1  tags after filtering in treatment: 48047 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1  Redundant rate of treatment: 0.03 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1  total tags in control: 50837 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 user defined the maximum tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1  tags after filtering in control: 50783 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1  Redundant rate of control: 0.00 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #1 finished! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 Build Peak Model... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 looking for paired plus/minus strand peaks... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 Total number of paired peaks: 1036 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 Model building with cross-correlation: Done </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 finished! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 predicted fragment length is 230 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2 alternative fragment length(s) may be 230,238 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #2.2 Generate R script for model : /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_broad_model.r </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Call peaks... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Pre-compute pvalue-qvalue table... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 In the peak calling step, the following will be performed simultaneously: </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... run_callpeak_broad_treat_pileup.bdg </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Write bedGraph files for control lambda (after scaling if necessary)... run_callpeak_broad_control_lambda.bdg </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Call peaks for each chromosome... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:26: #4 Write output xls file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_broad_peaks.xls </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:26: #4 Write broad peak in broadPeak format file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_broad_peaks.broadPeak </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:26: #4 Write broad peak in bed12/gappedPeak format file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_broad_peaks.gappedPeak </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:26: Done!</span></code></pre></div>
<p>Here are the outputs.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cp1</span>
<span class="co">#&gt; macsList of length 3</span>
<span class="co">#&gt; names(3): arguments outputs log</span>
<span class="va">cp2</span>
<span class="co">#&gt; macsList of length 3</span>
<span class="co">#&gt; names(3): arguments outputs log</span></code></pre></div>
</div>
<div id="the-macslist-class" class="section level2">
<h2 class="hasAnchor">
<a href="#the-macslist-class" class="anchor"></a>The <code>macsList</code> class</h2>
<p>The <code>macsList</code> is designed to contain everything of an execution, including function, inputs, outputs and logs, for the purpose of reproducibility.</p>
<p>For example, we can the function and input arguments.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cp1</span><span class="op">$</span><span class="va">arguments</span>
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; callpeak</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $tfile</span>
<span class="co">#&gt; CHIP</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $cfile</span>
<span class="co">#&gt; CTRL</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gsize</span>
<span class="co">#&gt; [1] 5.2e+07</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $outdir</span>
<span class="co">#&gt; tempdir()</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $name</span>
<span class="co">#&gt; [1] "run_callpeak_narrow0"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $store_bdg</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $cutoff_analysis</span>
<span class="co">#&gt; T</span></code></pre></div>
<p>The files of all the outputs are collected.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cp1</span><span class="op">$</span><span class="va">outputs</span>
<span class="co">#&gt; [1] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_control_lambda.bdg" </span>
<span class="co">#&gt; [2] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_cutoff_analysis.txt"</span>
<span class="co">#&gt; [3] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_model.r"            </span>
<span class="co">#&gt; [4] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_peaks.narrowPeak"   </span>
<span class="co">#&gt; [5] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_peaks.xls"          </span>
<span class="co">#&gt; [6] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_summits.bed"        </span>
<span class="co">#&gt; [7] "/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_treat_pileup.bdg"</span></code></pre></div>
<p>The <code>log</code> is especially important for <code>MACS</code> to check. Detailed information was given in the log when running.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">cp1</span><span class="op">$</span><span class="va">log</span><span class="op">)</span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: </span>
<span class="co">#&gt; # Command line: </span>
<span class="co">#&gt; # ARGUMENTS LIST:</span>
<span class="co">#&gt; # name = run_callpeak_narrow0</span>
<span class="co">#&gt; # format = AUTO</span>
<span class="co">#&gt; # ChIP-seq file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c68423d6b_4601']</span>
<span class="co">#&gt; # control file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c54ce7146_4606']</span>
<span class="co">#&gt; # effective genome size = 5.20e+07</span>
<span class="co">#&gt; # band width = 300</span>
<span class="co">#&gt; # model fold = [5.0, 50.0]</span>
<span class="co">#&gt; # qvalue cutoff = 5.00e-02</span>
<span class="co">#&gt; # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="co">#&gt; # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="co">#&gt; # Larger dataset will be scaled towards smaller dataset.</span>
<span class="co">#&gt; # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="co">#&gt; # Broad region calling is off</span>
<span class="co">#&gt; # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="co">#&gt; # Paired-End mode is off</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 read tag files... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 read treatment tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: Detected format is: BED </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: * Input file is gzipped. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1.2 read input tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: Detected format is: BED </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: * Input file is gzipped. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 tag size is determined as 101 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 tag size = 101.0 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  total tags in treatment: 49622 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 user defined the maximum tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  tags after filtering in treatment: 48047 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  Redundant rate of treatment: 0.03 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  total tags in control: 50837 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 user defined the maximum tags... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  tags after filtering in control: 50783 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1  Redundant rate of control: 0.00 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #1 finished! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 Build Peak Model... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 looking for paired plus/minus strand peaks... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 Total number of paired peaks: 1036 </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 Model building with cross-correlation: Done </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 finished! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 predicted fragment length is 230 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2 alternative fragment length(s) may be 230,238 bps </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #2.2 Generate R script for model : /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_model.r </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #3 Call peaks... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #3 Pre-compute pvalue-qvalue table... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:24: #3 Cutoff vs peaks called will be analyzed! </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_cutoff_analysis.txt' </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 In the peak calling step, the following will be performed simultaneously: </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... run_callpeak_narrow0_treat_pileup.bdg </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Write bedGraph files for control lambda (after scaling if necessary)... run_callpeak_narrow0_control_lambda.bdg </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3   Pileup will be based on sequencing depth in treatment. </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #3 Call peaks for each chromosome... </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #4 Write output xls file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_peaks.xls </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #4 Write peak in narrowPeak format file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_peaks.narrowPeak </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: #4 Write summits bed file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//Rtmp0zqyiO/run_callpeak_narrow0_summits.bed </span>
<span class="co">#&gt; INFO  @ Sat, 12 Dec 2020 00:39:25: Done!</span></code></pre></div>
</div>
</div>
<div id="resources" class="section level1">
<h1 class="hasAnchor">
<a href="#resources" class="anchor"></a>Resources</h1>
<p>More details about <code>MACS3</code> can be found: <a href="https://macs3-project.github.io/MACS/" class="uri">https://macs3-project.github.io/MACS/</a>.</p>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2020-12-09 r79601)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin13.4.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 11.0.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /Users/qi28068/miniconda3/envs/r-dev/lib/libopenblasp-r0.3.12.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] MACSdata_0.99.0  MACSr_0.99.5     BiocStyle_2.19.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.5                    lattice_0.20-41              </span>
<span class="co">#&gt;  [3] assertthat_0.2.1              rprojroot_2.0.2              </span>
<span class="co">#&gt;  [5] digest_0.6.27                 mime_0.9                     </span>
<span class="co">#&gt;  [7] BiocFileCache_1.15.1          R6_2.5.0                     </span>
<span class="co">#&gt;  [9] stats4_4.1.0                  RSQLite_2.2.1                </span>
<span class="co">#&gt; [11] evaluate_0.14                 httr_1.4.2                   </span>
<span class="co">#&gt; [13] pillar_1.4.7                  basilisk_1.3.1               </span>
<span class="co">#&gt; [15] rlang_0.4.9                   curl_4.3                     </span>
<span class="co">#&gt; [17] blob_1.2.1                    S4Vectors_0.29.6             </span>
<span class="co">#&gt; [19] Matrix_1.2-18                 reticulate_1.18              </span>
<span class="co">#&gt; [21] rmarkdown_2.5                 pkgdown_1.6.1                </span>
<span class="co">#&gt; [23] textshaping_0.2.1             desc_1.2.0                   </span>
<span class="co">#&gt; [25] AnnotationHub_2.23.0          stringr_1.4.0                </span>
<span class="co">#&gt; [27] bit_4.0.4                     shiny_1.5.0                  </span>
<span class="co">#&gt; [29] compiler_4.1.0                httpuv_1.5.4                 </span>
<span class="co">#&gt; [31] xfun_0.19                     pkgconfig_2.0.3              </span>
<span class="co">#&gt; [33] systemfonts_0.3.2             BiocGenerics_0.37.0          </span>
<span class="co">#&gt; [35] htmltools_0.5.0               tidyselect_1.1.0             </span>
<span class="co">#&gt; [37] tibble_3.0.4                  interactiveDisplayBase_1.29.0</span>
<span class="co">#&gt; [39] bookdown_0.21                 IRanges_2.25.4               </span>
<span class="co">#&gt; [41] withr_2.3.0                   crayon_1.3.4                 </span>
<span class="co">#&gt; [43] dplyr_1.0.2                   dbplyr_2.0.0                 </span>
<span class="co">#&gt; [45] later_1.1.0.1                 rappdirs_0.3.1               </span>
<span class="co">#&gt; [47] basilisk.utils_1.3.1          grid_4.1.0                   </span>
<span class="co">#&gt; [49] xtable_1.8-4                  jsonlite_1.7.2               </span>
<span class="co">#&gt; [51] lifecycle_0.2.0               DBI_1.1.0                    </span>
<span class="co">#&gt; [53] magrittr_2.0.1                stringi_1.5.3                </span>
<span class="co">#&gt; [55] fs_1.5.0                      promises_1.1.1               </span>
<span class="co">#&gt; [57] ellipsis_0.3.1                filelock_1.0.2               </span>
<span class="co">#&gt; [59] ragg_0.4.0                    generics_0.1.0               </span>
<span class="co">#&gt; [61] vctrs_0.3.5                   tools_4.1.0                  </span>
<span class="co">#&gt; [63] bit64_4.0.5                   Biobase_2.51.0               </span>
<span class="co">#&gt; [65] glue_1.4.2                    BiocVersion_3.13.1           </span>
<span class="co">#&gt; [67] purrr_0.3.4                   fastmap_1.0.1                </span>
<span class="co">#&gt; [69] parallel_4.1.0                yaml_2.2.1                   </span>
<span class="co">#&gt; [71] AnnotationDbi_1.53.0          BiocManager_1.30.10          </span>
<span class="co">#&gt; [73] ExperimentHub_1.17.0          memoise_1.1.0                </span>
<span class="co">#&gt; [75] knitr_1.30</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
