<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>bdgdiff — bdgdiff • MACSr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="bdgdiff — bdgdiff"><meta property="og:description" content="Differential peak detection based on paired four bedgraph
files. Note: All regions on the same chromosome in the bedGraph
file should be continuous so only bedGraph files from MACS3 are
accpetable."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/macs3-project/MACSr" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>bdgdiff</h1>
    
    <div class="hidden name"><code>bdgdiff.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Differential peak detection based on paired four bedgraph
files. Note: All regions on the same chromosome in the bedGraph
file should be continuous so only bedGraph files from MACS3 are
accpetable.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bdgdiff</span><span class="op">(</span></span>
<span>  <span class="va">t1bdg</span>,</span>
<span>  <span class="va">t2bdg</span>,</span>
<span>  <span class="va">c1bdg</span>,</span>
<span>  <span class="va">c2bdg</span>,</span>
<span>  cutoff <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  minlen <span class="op">=</span> <span class="fl">200L</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="fl">100L</span>,</span>
<span>  depth1 <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  depth2 <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  oprefix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  outputfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  log <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">2L</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>t1bdg</dt>
<dd><p>MACS pileup bedGraph for condition 1. Incompatible
with callpeak --SPMR output. REQUIRED</p></dd>


<dt>t2bdg</dt>
<dd><p>MACS pileup bedGraph for condition 2. Incompatible
with callpeak --SPMR output. REQUIRED</p></dd>


<dt>c1bdg</dt>
<dd><p>MACS control lambda bedGraph for condition
1. Incompatible with callpeak --SPMR output. REQUIRED</p></dd>


<dt>c2bdg</dt>
<dd><p>MACS control lambda bedGraph for condition
2. Incompatible with callpeak --SPMR output. REQUIRED</p></dd>


<dt>cutoff</dt>
<dd><p>logLR cutoff. DEFAULT: 3 (likelihood
ratio=1000)", default = 3</p></dd>


<dt>minlen</dt>
<dd><p>Minimum length of differential region. Try bigger value to remove small regions. DEFAULT: 200",
default = 200</p></dd>


<dt>maxgap</dt>
<dd><p>Maximum gap to merge nearby differential
regions. Consider a wider gap for broad marks. Maximum gap
should be smaller than minimum length (-g). DEFAULT:
100", default = 100</p></dd>


<dt>depth1</dt>
<dd><p>Sequencing depth (# of non-redundant reads in
million) for condition 1. It will be used together with
--d2. See description for --d2 below for how to assign
them. Default: 1</p></dd>


<dt>depth2</dt>
<dd><p>Sequencing depth (# of non-redundant reads in
million) for condition 2. It will be used together with
--d1. DEPTH1 and DEPTH2 will be used to calculate scaling
factor for each sample, to down-scale larger sample to the
level of smaller one. For example, while comparing 10 million
condition 1 and 20 million condition 2, use --d1 10 --d2 20,
then pileup value in bedGraph for condition 2 will be divided
by 2. Default: 1</p></dd>


<dt>outdir</dt>
<dd><p>The output directory.</p></dd>


<dt>oprefix</dt>
<dd><p>Output file prefix. Actual files will be named as
PREFIX_cond1.bed, PREFIX_cond2.bed and
PREFIX_common.bed. Mutually exclusive with -o/--ofile.</p></dd>


<dt>outputfile</dt>
<dd><p>Output filenames. Must give three arguments in
order: 1. file for unique regions in condition 1; 2. file for
unique regions in condition 2; 3. file for common regions in
both conditions. Note: mutually exclusive with --o-prefix.</p></dd>


<dt>log</dt>
<dd><p>Whether to capture logs.</p></dd>


<dt>verbose</dt>
<dd><p>Set verbose level of runtime message. 0: only show
critical message, 1: show additional warning message, 2: show
process information, 3: show debug messages. DEFAULT:2</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><code>macsList</code> object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">CHIP</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4558"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">CTRL</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4563"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">c1</span> <span class="op">&lt;-</span> <span class="fu"><a href="callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>, cutoff_analysis <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>               outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"callpeak_narrow0"</span>, store_bdg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [729 MB] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Command line: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ARGUMENTS LIST:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # name = callpeak_narrow0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # format = AUTO</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # effective genome size = 5.20e+07</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # band width = 300</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # model fold = [5.0, 50.0]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # qvalue cutoff = 5.00e-02</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Larger dataset will be scaled towards smaller dataset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Broad region calling is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Paired-End mode is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [729 MB] #1 read tag files... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [729 MB] #1 read treatment tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [732 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [732 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [733 MB] #1.2 read input tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [733 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [733 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 tag size is determined as 101 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 tag size = 101.0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1  total tags in treatment: 49622 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1  tags after filtering in treatment: 48047 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1  Redundant rate of treatment: 0.03 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1  total tags in control: 50837 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1  tags after filtering in control: 50783 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1  Redundant rate of control: 0.00 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #1 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 Build Peak Model... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 looking for paired plus/minus strand peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 Total number of paired peaks: 469 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 Model building with cross-correlation: Done </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 predicted fragment length is 228 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2 alternative fragment length(s) may be 228 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #2.2 Generate R script for model : /tmp/RtmpEPj3Vo/callpeak_narrow0_model.r </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #3 Call peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #3 Pre-compute pvalue-qvalue table... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:13: [734 MB] #3 Cutoff vs peaks called will be analyzed! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/tmp/RtmpEPj3Vo/callpeak_narrow0_cutoff_analysis.txt' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #3 In the peak calling step, the following will be performed simultaneously: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #3   Write bedGraph files for treatment pileup (after scaling if necessary)... callpeak_narrow0_treat_pileup.bdg </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #3   Write bedGraph files for control lambda (after scaling if necessary)... callpeak_narrow0_control_lambda.bdg </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #3   Pileup will be based on sequencing depth in treatment. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #3 Call peaks for each chromosome... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #4 Write output xls file... /tmp/RtmpEPj3Vo/callpeak_narrow0_peaks.xls </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #4 Write peak in narrowPeak format file... /tmp/RtmpEPj3Vo/callpeak_narrow0_peaks.narrowPeak </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #4 Write summits bed file... /tmp/RtmpEPj3Vo/callpeak_narrow0_summits.bed </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] Done! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">c2</span> <span class="op">&lt;-</span> <span class="fu"><a href="callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">1e7</span>, nomodel <span class="op">=</span> <span class="cn">TRUE</span>, extsize <span class="op">=</span> <span class="fl">250</span>,</span></span>
<span class="r-in"><span>               outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"callpeak_narrow_revert"</span>, store_bdg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Command line: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ARGUMENTS LIST:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # name = callpeak_narrow_revert</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # format = AUTO</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # effective genome size = 1.00e+07</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # band width = 300</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # model fold = [5.0, 50.0]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # qvalue cutoff = 5.00e-02</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Larger dataset will be scaled towards smaller dataset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Broad region calling is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Paired-End mode is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #1 read tag files... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] #1 read treatment tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [757 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1.2 read input tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 tag size is determined as 101 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 tag size = 101.0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1  total tags in treatment: 49622 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1  tags after filtering in treatment: 48047 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1  Redundant rate of treatment: 0.03 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1  total tags in control: 50837 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1  tags after filtering in control: 50783 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1  Redundant rate of control: 0.00 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #1 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #2 Build Peak Model... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #2 Skipped... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #2 Use 250 as fragment length </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3 Call peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3 Pre-compute pvalue-qvalue table... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3 In the peak calling step, the following will be performed simultaneously: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3   Write bedGraph files for treatment pileup (after scaling if necessary)... callpeak_narrow_revert_treat_pileup.bdg </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3   Write bedGraph files for control lambda (after scaling if necessary)... callpeak_narrow_revert_control_lambda.bdg </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3   Pileup will be based on sequencing depth in treatment. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #3 Call peaks for each chromosome... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #4 Write output xls file... /tmp/RtmpEPj3Vo/callpeak_narrow_revert_peaks.xls </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #4 Write peak in narrowPeak format file... /tmp/RtmpEPj3Vo/callpeak_narrow_revert_peaks.narrowPeak </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] #4 Write summits bed file... /tmp/RtmpEPj3Vo/callpeak_narrow_revert_summits.bed </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] Done! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">t1bdg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"treat_pileup"</span>, <span class="va">c1</span><span class="op">$</span><span class="va">outputs</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">c1bdg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"control_lambda"</span>, <span class="va">c1</span><span class="op">$</span><span class="va">outputs</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">t2bdg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"treat_pileup"</span>, <span class="va">c2</span><span class="op">$</span><span class="va">outputs</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">c2bdg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"control_lambda"</span>, <span class="va">c2</span><span class="op">$</span><span class="va">outputs</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">bdgdiff</span><span class="op">(</span><span class="va">t1bdg</span>, <span class="va">t2bdg</span>, <span class="va">c1bdg</span>, <span class="va">c2bdg</span>,</span></span>
<span class="r-in"><span>        outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, oprefix <span class="op">=</span> <span class="st">"bdgdiff"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] Read and build treatment 1 bedGraph... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] Read and build control 1 bedGraph... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:14: [758 MB] Read and build treatment 2 bedGraph... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:15: [758 MB] Read and build control 2 bedGraph... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:15: [758 MB] Write peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:15: [758 MB] Done </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> macsList class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  /tmp/RtmpEPj3Vo/bdgdiff_c3.0_common.bed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  /tmp/RtmpEPj3Vo/bdgdiff_c3.0_cond1.bed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  /tmp/RtmpEPj3Vo/bdgdiff_c3.0_cond2.bed </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $arguments: t1bdg, t2bdg, c1bdg, c2bdg, outdir, oprefix </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $log:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:14: [758 MB] Read and build treatment 1 bedGraph... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:14: [758 MB] Read and build control 1 bedGraph... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:14: [758 MB] Read and build treatment 2 bedGraph... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:15: [758 MB] Read and build control 2 bedGraph... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:15: [758 MB] Write peaks... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

