<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>bdgopt — bdgopt • MACSr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="bdgopt — bdgopt"><meta property="og:description" content="Operations on score column of bedGraph file. Note: All regions on
the same chromosome in the bedGraph file should be continuous so
only bedGraph files from MACS3 are accpetable."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/macs3-project/MACSr" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>bdgopt</h1>
    
    <div class="hidden name"><code>bdgopt.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Operations on score column of bedGraph file. Note: All regions on
the same chromosome in the bedGraph file should be continuous so
only bedGraph files from MACS3 are accpetable.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bdgopt</span><span class="op">(</span></span>
<span>  <span class="va">ifile</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"multiply"</span>, <span class="st">"add"</span>, <span class="st">"p2q"</span>, <span class="st">"max"</span>, <span class="st">"min"</span><span class="op">)</span>,</span>
<span>  extraparam <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  outputfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  log <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">2L</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ifile</dt>
<dd><p>MACS score in bedGraph. Note: this must be a bedGraph
file covering the ENTIRE genome. REQUIRED</p></dd>


<dt>method</dt>
<dd><p>Method to modify the score column of bedGraph
file. Available choices are: multiply, add, max, min, or
p2q. 1) multiply, the EXTRAPARAM is required and will be
multiplied to the score column. If you intend to divide the
score column by X, use value of 1/X as EXTRAPARAM. 2) add, the
EXTRAPARAM is required and will be added to the score
column. If you intend to subtract the score column by X, use
value of -X as EXTRAPARAM. 3) max, the EXTRAPARAM is required
and will take the maximum value between score and the
EXTRAPARAM. 4) min, the EXTRAPARAM is required and will take
the minimum value between score and the EXTRAPARAM. 5) p2q,
this will convert p-value scores to q-value scores using
Benjamini-Hochberg process. The EXTRAPARAM is not
required. This method assumes the scores are -log10 p-value
from MACS3. Any other types of score will cause unexpected
errors.", default="p2q"</p></dd>


<dt>extraparam</dt>
<dd><p>The extra parameter for METHOD. Check the detail
of -m option.</p></dd>


<dt>outputfile</dt>
<dd><p>Output filename. Mutually exclusive with
--o-prefix. The number and the order of arguments for --ofile
must be the same as for -m.</p></dd>


<dt>outdir</dt>
<dd><p>The output directory.</p></dd>


<dt>log</dt>
<dd><p>Whether to capture logs.</p></dd>


<dt>verbose</dt>
<dd><p>Set verbose level of runtime message. 0: only show
critical message, 1: show additional warning message, 2: show
process information, 3: show debug messages. DEFAULT:2</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><code>macsList</code> object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">CHIP</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4558"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">CTRL</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4563"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">c1</span> <span class="op">&lt;-</span> <span class="fu"><a href="callpeak.html">callpeak</a></span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>, cutoff_analysis <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>               outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"callpeak_narrow0"</span>,</span></span>
<span class="r-in"><span>               store_bdg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Command line: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ARGUMENTS LIST:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # name = callpeak_narrow0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # format = AUTO</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # effective genome size = 5.20e+07</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # band width = 300</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # model fold = [5.0, 50.0]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # qvalue cutoff = 5.00e-02</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Larger dataset will be scaled towards smaller dataset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Broad region calling is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Paired-End mode is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 read tag files... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 read treatment tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1.2 read input tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 tag size is determined as 101 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 tag size = 101.0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1  total tags in treatment: 49622 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1  tags after filtering in treatment: 48047 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1  Redundant rate of treatment: 0.03 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1  total tags in control: 50837 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1  tags after filtering in control: 50783 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1  Redundant rate of control: 0.00 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #1 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 Build Peak Model... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 looking for paired plus/minus strand peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 Total number of paired peaks: 469 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 Model building with cross-correlation: Done </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 predicted fragment length is 228 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2 alternative fragment length(s) may be 228 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #2.2 Generate R script for model : /tmp/RtmpEPj3Vo/callpeak_narrow0_model.r </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #3 Call peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #3 Pre-compute pvalue-qvalue table... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:18: [758 MB] #3 Cutoff vs peaks called will be analyzed! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/tmp/RtmpEPj3Vo/callpeak_narrow0_cutoff_analysis.txt' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #3 In the peak calling step, the following will be performed simultaneously: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #3   Write bedGraph files for treatment pileup (after scaling if necessary)... callpeak_narrow0_treat_pileup.bdg </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #3   Write bedGraph files for control lambda (after scaling if necessary)... callpeak_narrow0_control_lambda.bdg </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #3   Pileup will be based on sequencing depth in treatment. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #3 Call peaks for each chromosome... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #4 Write output xls file... /tmp/RtmpEPj3Vo/callpeak_narrow0_peaks.xls </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #4 Write peak in narrowPeak format file... /tmp/RtmpEPj3Vo/callpeak_narrow0_peaks.narrowPeak </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] #4 Write summits bed file... /tmp/RtmpEPj3Vo/callpeak_narrow0_summits.bed </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] Done! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">cfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"treat_pileup.bdg"</span>, <span class="va">c1</span><span class="op">$</span><span class="va">outputs</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">bdgopt</span><span class="op">(</span><span class="va">cfile</span>, method <span class="op">=</span> <span class="st">"min"</span>, extraparam <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>       outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, outputfile <span class="op">=</span> <span class="st">"bdgopt_min.bdg"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] Read and build bedGraph... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] Modify bedGraph... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] Write bedGraph of modified scores... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 13:42:19: [758 MB] Finished 'min'! Please check '/tmp/RtmpEPj3Vo/bdgopt_min.bdg'! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> macsList class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  /tmp/RtmpEPj3Vo/bdgopt_min.bdg </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $arguments: ifile, method, extraparam, outputfile, outdir </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $log:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:19: [758 MB] Read and build bedGraph... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:19: [758 MB] Modify bedGraph... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:19: [758 MB] Write bedGraph of modified scores... </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  INFO  @ 17 Nov 2023 13:42:19: [758 MB] Finished 'min'! Please check '/tmp/RtmpEPj3Vo/bdgopt_min.bdg'! </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

