<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>callpeak — callpeak • MACSr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="callpeak — callpeak"><meta property="og:description" content="Main MACS3 Function to call peaks from alignment results."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/macs3-project/MACSr" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>callpeak</h1>
    
    <div class="hidden name"><code>callpeak.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Main MACS3 Function to call peaks from alignment results.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">callpeak</span><span class="op">(</span></span>
<span>  <span class="va">tfile</span>,</span>
<span>  cfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gsize <span class="op">=</span> <span class="st">"hs"</span>,</span>
<span>  tsize <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"AUTO"</span>,</span>
<span>  keepduplicates <span class="op">=</span> <span class="st">"1"</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"NA"</span>,</span>
<span>  store_bdg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  do_SPMR <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  trackline <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nomodel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  shift <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  extsize <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  bw <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  d_min <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  mfold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">50</span><span class="op">)</span>,</span>
<span>  onauto <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  qvalue <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  pvalue <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tempdir <span class="op">=</span> <span class="st">"/tmp"</span>,</span>
<span>  nolambda <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  scaleto <span class="op">=</span> <span class="st">"small"</span>,</span>
<span>  downsample <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  slocal <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  llocal <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  broad <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  broadcutoff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minlen <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cutoff_analysis <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  fecutoff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  call_summits <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  buffer_size <span class="op">=</span> <span class="fl">1e+05</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>  log <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>tfile</dt>
<dd><p>ChIP-seq treatment files.</p></dd>


<dt>cfile</dt>
<dd><p>Control files.</p></dd>


<dt>gsize</dt>
<dd><p>Effective genome size. It can be 1.0e+9 or 1000000000,
or shortcuts:'hs' for human (2.7e9), 'mm' for mouse (1.87e9),
'ce' for C. elegans (9e7) and 'dm' for fruitfly (1.2e8),
Default:hs.</p></dd>


<dt>tsize</dt>
<dd><p>Tag size/read length. This will override the auto
detected tag size. DEFAULT: Not set</p></dd>


<dt>format</dt>
<dd><p>Format of tag file, "AUTO", "BED" or "ELAND" or
"ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or "BOWTIE" or
"BAMPE" or "BEDPE".</p></dd>


<dt>keepduplicates</dt>
<dd><p>It controls the behavior towards duplicate
tags at the exact same location -- the same coordination and
the same strand.</p></dd>


<dt>outdir</dt>
<dd><p>If specified all output files will be written to that
directory.</p></dd>


<dt>name</dt>
<dd><p>Experiment name, which will be used to generate output
file names.</p></dd>


<dt>store_bdg</dt>
<dd><p>Whether or not to save extended fragment pileup,
and local lambda tracks (two files) at every bp into a bedGraph
file.</p></dd>


<dt>do_SPMR</dt>
<dd><p>If True, MACS will SAVE signal per million reads for
fragment pileup profiles.</p></dd>


<dt>trackline</dt>
<dd><p>Tells MACS to include trackline with bedGraph
files.</p></dd>


<dt>nomodel</dt>
<dd><p>Whether or not to build the shifting model.</p></dd>


<dt>shift</dt>
<dd><p>The arbitrary shift in bp. Use discretion while
setting it other than default value.</p></dd>


<dt>extsize</dt>
<dd><p>The arbitrary extension size in bp.</p></dd>


<dt>bw</dt>
<dd><p>Band width for picking regions to compute fragment size.</p></dd>


<dt>d_min</dt>
<dd><p>Minimum fragment size in basepair. Any predicted
fragment size less than this will be excluded.</p></dd>


<dt>mfold</dt>
<dd><p>Select the regions within MFOLD range of
high-confidence enrichment ratio against background to build
model.</p></dd>


<dt>onauto</dt>
<dd><p>Whether turn on the auto pair model process.</p></dd>


<dt>qvalue</dt>
<dd><p>Minimum FDR (q-value) cutoff for peak detection.</p></dd>


<dt>pvalue</dt>
<dd><p>Pvalue cutoff for peak detection. DEFAULT: not set.</p></dd>


<dt>tempdir</dt>
<dd><p>Optional directory to store temp files.</p></dd>


<dt>nolambda</dt>
<dd><p>If True, MACS will use fixed background lambda as
local lambda for every peak region.</p></dd>


<dt>scaleto</dt>
<dd><p>When set to 'small', scale the larger sample up to
the smaller sample.</p></dd>


<dt>downsample</dt>
<dd><p>When set, random sampling method will scale down
the bigger sample. By default, MACS uses linear scaling.</p></dd>


<dt>slocal</dt>
<dd><p>The small nearby region in basepairs to calculate
dynamic lambda.</p></dd>


<dt>llocal</dt>
<dd><p>The large nearby region in basepairs to calculate
dynamic lambda.</p></dd>


<dt>broad</dt>
<dd><p>If set, MACS will try to call broad peaks using the
--broad-cutoff setting.</p></dd>


<dt>broadcutoff</dt>
<dd><p>Cutoff for broad region. This option is not
available unless --broad is set.</p></dd>


<dt>maxgap</dt>
<dd><p>Maximum gap between significant sites to cluster them
together. The DEFAULT value is the detected read length/tag
size.</p></dd>


<dt>minlen</dt>
<dd><p>Minimum length of a peak. The DEFAULT value is the predicted
fragment size d.</p></dd>


<dt>cutoff_analysis</dt>
<dd><p>While set, MACS2 will analyze number or
total length of peaks that can be called by different p-value
cutoff then output a summary table to help user decide a better
cutoff.</p></dd>


<dt>fecutoff</dt>
<dd><p>When set, the value will be used to filter out
peaks with low fold-enrichment.</p></dd>


<dt>call_summits</dt>
<dd><p>If set, MACS will use a more sophisticated
signal processing approach to find subpeak summits in each
enriched peak region.</p></dd>


<dt>buffer_size</dt>
<dd><p>Buffer size for incrementally increasing
internal array size to store reads alignment
information. DEFAULT: 100000.</p></dd>


<dt>verbose</dt>
<dd><p>Set verbose level of runtime message. 0: only show
critical message, 1: show additional warning message, 2: show
process information, 3: show debug messages. DEFAULT:2</p></dd>


<dt>log</dt>
<dd><p>Whether to capture logs.</p></dd>


<dt>...</dt>
<dd><p>More options for macs2.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><code>macsList</code> object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">CHIP</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4558"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">CTRL</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH4563"</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?MACSdata and browseVignettes('MACSdata') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">callpeak</span><span class="op">(</span><span class="va">CHIP</span>, <span class="va">CTRL</span>, gsize <span class="op">=</span> <span class="fl">5.2e7</span>,</span></span>
<span class="r-in"><span>                cutoff_analysis <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                name <span class="op">=</span> <span class="st">"callpeak_narrow0"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Command line: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ARGUMENTS LIST:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # name = callpeak_narrow0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # format = AUTO</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # ChIP-seq file = ['/home/qhu/.cache/R/ExperimentHub/1071e4528bfa6_4601']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # control file = ['/home/qhu/.cache/R/ExperimentHub/1071e71fe08e5_4606']</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # effective genome size = 5.20e+07</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # band width = 300</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # model fold = [5.0, 50.0]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # qvalue cutoff = 5.00e-02</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The maximum gap between significant sites is assigned as the read length/tag size.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # The minimum length of peaks is assigned as the predicted fragment length "d".</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Larger dataset will be scaled towards smaller dataset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Broad region calling is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Additional cutoff on fold-enrichment is: 0.10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # Paired-End mode is off</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 read tag files... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 read treatment tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1.2 read input tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] Detected format is: BED </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] * Input file is gzipped. </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 tag size is determined as 101 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 tag size = 101.0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1  total tags in treatment: 49622 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1  tags after filtering in treatment: 48047 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1  Redundant rate of treatment: 0.03 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1  total tags in control: 50837 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 user defined the maximum tags... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1  tags after filtering in control: 50783 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1  Redundant rate of control: 0.00 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #1 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 Build Peak Model... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 looking for paired plus/minus strand peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 Total number of paired peaks: 469 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 Model building with cross-correlation: Done </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 finished! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 predicted fragment length is 228 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2 alternative fragment length(s) may be 228 bps </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #2.2 Generate R script for model : /tmp/RtmpeMHyqk/callpeak_narrow0_model.r </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #3 Call peaks... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #3 Pre-compute pvalue-qvalue table... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:26: [758 MB] #3 Cutoff vs peaks called will be analyzed! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:27: [761 MB] #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/tmp/RtmpeMHyqk/callpeak_narrow0_cutoff_analysis.txt' </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:27: [761 MB] #3 Call peaks for each chromosome... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:27: [761 MB] #4 Write output xls file... /tmp/RtmpeMHyqk/callpeak_narrow0_peaks.xls </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:27: [761 MB] #4 Write peak in narrowPeak format file... /tmp/RtmpeMHyqk/callpeak_narrow0_peaks.narrowPeak </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:27: [761 MB] #4 Write summits bed file... /tmp/RtmpeMHyqk/callpeak_narrow0_summits.bed </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> INFO  @ 17 Nov 2023 12:41:27: [761 MB] Done! </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

