<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>callvar — callvar • MACSr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="callvar — callvar"><meta property="og:description" content="Call variants in given peak regions from the alignment BAM files."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/macs3-project/MACSr" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>callvar</h1>
    
    <div class="hidden name"><code>callvar.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Call variants in given peak regions from the alignment BAM files.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">callvar</span><span class="op">(</span></span>
<span>  <span class="va">peakbed</span>,</span>
<span>  <span class="va">tfile</span>,</span>
<span>  <span class="va">cfile</span>,</span>
<span>  outputfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  GQCutoffHetero <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  GQCutoffHomo <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  Q <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  maxDuplicate <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  fermi <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  fermiMinOverlap <span class="op">=</span> <span class="fl">30L</span>,</span>
<span>  top2allelesMinRatio <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  altalleleMinCount <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>  maxAR <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  np <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>  log <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>peakbed</dt>
<dd><p>Peak regions in BED format, sorted by
coordinates. REQUIRED.</p></dd>


<dt>tfile</dt>
<dd><p>ChIP-seq/ATAC-seq treatment file in BAM format,
containing only records in peak regions, sorted by
coordinates. Check instruction on how to make the file using
samtools. REQUIRED.</p></dd>


<dt>cfile</dt>
<dd><p>Control file in BAM format, containing only records in
peak regions, sorted by coordinates. Check instruction on how
to make the file using samtools.</p></dd>


<dt>outputfile</dt>
<dd><p>Output VCF file name.</p></dd>


<dt>GQCutoffHetero</dt>
<dd><p>Genotype Quality score
(-10log10((L00+L11)/(L01+L00+L11))) cutoff for Heterozygous
allele type. Default:0, or there is no cutoff on GQ.</p></dd>


<dt>GQCutoffHomo</dt>
<dd><p>Genotype Quality score
(-10log10((L00+L01)/(L01+L00+L11))) cutoff for Homozygous
allele (not the same as reference) type. Default:0, or ther is
no cutoff on GQ.</p></dd>


<dt>Q</dt>
<dd><p>Only consider bases with quality score greater than this
value. Default: 20, which means Q20 or 0.01 error rate.</p></dd>


<dt>maxDuplicate</dt>
<dd><p>Maximum duplicated reads allowed per mapping
position, mapping strand and the same CIGAR code. Default:
1. When sequencing depth is high, to set a higher value might
help evaluate the correct allele ratio.</p></dd>


<dt>fermi</dt>
<dd><p>Option to control when to apply local assembly through
Fermi. By default (set as 'auto'), while SAPPER detects any
INDEL variant in a peak region, it will utilize Fermi to
recover the actual DNA sequences to refine the read
alignments. If set as 'on', Fermi will be always invoked. It
can increase specificity however sensivity and speed will be
significantly lower. If set as 'off', Fermi won't be invoked at
all. If so, speed and sensitivity can be higher but specificity
will be significantly lower. Default: auto</p></dd>


<dt>fermiMinOverlap</dt>
<dd><p>The minimal overlap for fermi to initially
assemble two reads. Must be between 1 and read length. A longer
fermiMinOverlap is needed while read length is small (e.g. 30
for 36bp read, but 33 for 100bp read may work). Default:30</p></dd>


<dt>top2allelesMinRatio</dt>
<dd><p>The reads for the top 2 most frequent
alleles (e.g. a ref allele and an alternative allele) at a loci
shouldn't be too few comparing to total reads mapped. The
minimum ratio is set by this optoin. Must be a float between
0.5 and 1.  Default:0.8 which means at least 80%% of reads
contain the top 2 alleles.</p></dd>


<dt>altalleleMinCount</dt>
<dd><p>The count of the alternative
(non-reference) allele at a loci shouldn't be too few. By
default, we require at least two reads support the alternative
allele. Default:2</p></dd>


<dt>maxAR</dt>
<dd><p>The maximum Allele-Ratio allowed while calculating
likelihood for allele-specific binding. If we allow higher
maxAR, we may mistakenly assign some homozygous loci as
heterozygous. Default:0.95</p></dd>


<dt>np</dt>
<dd><p>CPU used for mutliple processing. Please note that,
assigning more CPUs does not guarantee the process being
faster. Creating too many parrallel processes need memory
operations and may negate benefit from multi
processing. Default: 1</p></dd>


<dt>verbose</dt>
<dd><p>Set verbose level of runtime message. 0: only show
critical message, 1: show additional warning message, 2: show
process information, 3: show debug messages. DEFAULT:2</p></dd>


<dt>log</dt>
<dd><p>Whether to capture logs.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><code>macsList</code> object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu">callvar</span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="st">"PEsample_peaks_sorted.bed"</span>,</span></span>
<span class="r-in"><span><span class="st">"PEsample_peaks_sorted.bam"</span>,</span></span>
<span class="r-in"><span><span class="st">"PEcontrol_peaks_sorted.bam"</span>,</span></span>
<span class="r-in"><span><span class="st">"/tmp/test.vcf"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

