<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>cmbreps — cmbreps • MACSr</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="cmbreps — cmbreps" />
<meta property="og:description" content="Combine BEDGraphs of scores from replicates. Note: All regions on
the same chromosome in the bedGraph file should be continuous so
only bedGraph files from MACS3 are accpetable." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/macs3-project/MACSr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>cmbreps</h1>
    
    <div class="hidden name"><code>cmbreps.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Combine BEDGraphs of scores from replicates. Note: All regions on
the same chromosome in the bedGraph file should be continuous so
only bedGraph files from MACS3 are accpetable.</p>
    </div>

    <pre class="usage"><span class='fu'>cmbreps</span><span class='op'>(</span>
  ifiles <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  weights <span class='op'>=</span> <span class='fl'>1</span>,
  method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"fisher"</span>, <span class='st'>"max"</span>, <span class='st'>"mean"</span><span class='op'>)</span>,
  outputfile <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/character.html'>character</a></span><span class='op'>(</span><span class='op'>)</span>,
  outdir <span class='op'>=</span> <span class='st'>"."</span>,
  log <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ifiles</th>
      <td><p>MACS score in bedGraph for each replicate. Require at
least 2 files such as '-i A B C D'. REQUIRED</p></td>
    </tr>
    <tr>
      <th>weights</th>
      <td><p>Weight for each replicate. Default is 1.0 for
each. When given, require same number of parameters as IFILE.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>to use while combining scores from replicates. 1)
fisher: Fisher's combined probability test. It requires scores
in ppois form (-log10 pvalues) from bdgcmp. Other types of
scores for this method may cause cmbreps unexpected errors. 2)
max: take the maximum value from replicates for each genomic
position. 3) mean: take the average value. Note, except for
Fisher's method, max or mean will take scores AS IS which means
they won't convert scores from log scale to linear scale or
vice versa.", default="fisher"</p></td>
    </tr>
    <tr>
      <th>outputfile</th>
      <td><p>Output filename. Mutually exclusive with
--o-prefix. The number and the order of arguments for --ofile
must be the same as for -m.</p></td>
    </tr>
    <tr>
      <th>outdir</th>
      <td><p>The output directory.</p></td>
    </tr>
    <tr>
      <th>log</th>
      <td><p>Whether to capture logs.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>macsList</code> object.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>eh</span> <span class='op'>&lt;-</span> <span class='fu'>ExperimentHub</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html'>ExperimentHub</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2020-12-09</span></div><div class='input'><span class='va'>CHIP</span> <span class='op'>&lt;-</span> <span class='va'>eh</span><span class='op'>[[</span><span class='st'>"EH4558"</span><span class='op'>]</span><span class='op'>]</span>
</div><div class='output co'>#&gt; <span class='message'>see ?MACSdata and browseVignettes('MACSdata') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>CTRL</span> <span class='op'>&lt;-</span> <span class='va'>eh</span><span class='op'>[[</span><span class='st'>"EH4563"</span><span class='op'>]</span><span class='op'>]</span>
</div><div class='output co'>#&gt; <span class='message'>see ?MACSdata and browseVignettes('MACSdata') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>c1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='callpeak.html'>callpeak</a></span><span class='op'>(</span><span class='va'>CHIP</span>, <span class='va'>CTRL</span>, gsize <span class='op'>=</span> <span class='fl'>5.2e7</span>, cutoff_analysis <span class='op'>=</span> <span class='cn'>TRUE</span>,
outdir <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>, name <span class='op'>=</span> <span class='st'>"callpeak_narrow0"</span>, store_bdg <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:22: </span>
#&gt; <span class='message'># Command line: </span>
#&gt; <span class='message'># ARGUMENTS LIST:</span>
#&gt; <span class='message'># name = callpeak_narrow0</span>
#&gt; <span class='message'># format = AUTO</span>
#&gt; <span class='message'># ChIP-seq file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c68423d6b_4601']</span>
#&gt; <span class='message'># control file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c54ce7146_4606']</span>
#&gt; <span class='message'># effective genome size = 5.20e+07</span>
#&gt; <span class='message'># band width = 300</span>
#&gt; <span class='message'># model fold = [5.0, 50.0]</span>
#&gt; <span class='message'># qvalue cutoff = 5.00e-02</span>
#&gt; <span class='message'># The maximum gap between significant sites is assigned as the read length/tag size.</span>
#&gt; <span class='message'># The minimum length of peaks is assigned as the predicted fragment length "d".</span>
#&gt; <span class='message'># Larger dataset will be scaled towards smaller dataset.</span>
#&gt; <span class='message'># Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
#&gt; <span class='message'># Broad region calling is off</span>
#&gt; <span class='message'># Additional cutoff on fold-enrichment is: 0.10</span>
#&gt; <span class='message'># Paired-End mode is off</span>
#&gt; <span class='message'> </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:22: #1 read tag files... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:22: #1 read treatment tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: Detected format is: BED </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: * Input file is gzipped. </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1.2 read input tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: Detected format is: BED </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: * Input file is gzipped. </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 tag size is determined as 101 bps </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 tag size = 101.0 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1  total tags in treatment: 49622 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 user defined the maximum tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1  tags after filtering in treatment: 48047 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1  Redundant rate of treatment: 0.03 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1  total tags in control: 50837 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 user defined the maximum tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1  tags after filtering in control: 50783 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1  Redundant rate of control: 0.00 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #1 finished! </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 Build Peak Model... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 looking for paired plus/minus strand peaks... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 Total number of paired peaks: 1036 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 Model building with cross-correlation: Done </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 finished! </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 predicted fragment length is 230 bps </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2 alternative fragment length(s) may be 230,238 bps </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #2.2 Generate R script for model : /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_model.r </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3 Call peaks... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3 Pre-compute pvalue-qvalue table... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3 Cutoff vs peaks called will be analyzed! </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_cutoff_analysis.txt' </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3 In the peak calling step, the following will be performed simultaneously: </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... callpeak_narrow0_treat_pileup.bdg </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3   Write bedGraph files for control lambda (after scaling if necessary)... callpeak_narrow0_control_lambda.bdg </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3   Pileup will be based on sequencing depth in treatment. </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #3 Call peaks for each chromosome... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #4 Write output xls file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_peaks.xls </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #4 Write peak in narrowPeak format file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_peaks.narrowPeak </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: #4 Write summits bed file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_summits.bed </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:23: Done! </span>
#&gt; <span class='message'></span></div><div class='output co'>#&gt; <span class='error'>Error in .setupMethodsTables(fdef, initialize = TRUE): trying to get slot "group" from an object of a basic class ("NULL") with no slots</span></div><div class='input'><span class='fu'>cmbreps</span><span class='op'>(</span>ifiles <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>c1</span><span class='op'>$</span><span class='va'>outputs</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span>, <span class='va'>c1</span><span class='op'>$</span><span class='va'>outputs</span><span class='op'>[</span><span class='fl'>7</span><span class='op'>]</span><span class='op'>)</span>,
method <span class='op'>=</span> <span class='st'>"max"</span>, outdir <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>, outputfile <span class='op'>=</span> <span class='st'>"cmbreps"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in cmbreps(ifiles = list(c1$outputs[1], c1$outputs[7]), method = "max",     outdir = tempdir(), outputfile = "cmbreps"): object 'c1' not found</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


