<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>refinepeak — refinepeak • MACSr</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="refinepeak — refinepeak" />
<meta property="og:description" content="(Experimental) Take raw reads alignment, refine peak summits and
give scores measuring balance of waston/crick tags. Inspired by
SPP." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MACSr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MACSr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/macs3-project/MACSr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>refinepeak</h1>
    
    <div class="hidden name"><code>refinepeak.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>(Experimental) Take raw reads alignment, refine peak summits and
give scores measuring balance of waston/crick tags. Inspired by
SPP.</p>
    </div>

    <pre class="usage"><span class='fu'>refinepeak</span><span class='op'>(</span>
  <span class='va'>bedfile</span>,
  <span class='va'>ifile</span>,
  format <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"AUTO"</span>, <span class='st'>"BAM"</span>, <span class='st'>"SAM"</span>, <span class='st'>"BED"</span>, <span class='st'>"ELAND"</span>, <span class='st'>"ELANDMULTI"</span>, <span class='st'>"ELANDEXPORT"</span>,
    <span class='st'>"BOWTIE"</span><span class='op'>)</span>,
  cutoff <span class='op'>=</span> <span class='fl'>5</span>,
  windowsize <span class='op'>=</span> <span class='fl'>200L</span>,
  buffer_size <span class='op'>=</span> <span class='fl'>100000L</span>,
  verbose <span class='op'>=</span> <span class='fl'>2L</span>,
  outdir <span class='op'>=</span> <span class='st'>"./"</span>,
  outputfile <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/character.html'>character</a></span><span class='op'>(</span><span class='op'>)</span>,
  log <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bedfile</th>
      <td><p>Candidate peak file in BED format. REQUIRED.</p></td>
    </tr>
    <tr>
      <th>ifile</th>
      <td><p>ChIP-seq alignment file. If multiple files are given
as '-t A B C', then they will all be read and combined. Note
that pair-end data is not supposed to work with this
command. REQUIRED.</p></td>
    </tr>
    <tr>
      <th>format</th>
      <td><p>Format of tag file, \"AUTO\", \"BED\" or \"ELAND\" or
\"ELANDMULTI\" or \"ELANDEXPORT\" or \"SAM\" or \"BAM\" or
\"BOWTIE\". The default AUTO option will let '%(prog)s' decide
which format the file is. Please check the definition in README
file if you choose
ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT: \"AUTO\""</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>Cutoff DEFAULT: 5</p></td>
    </tr>
    <tr>
      <th>windowsize</th>
      <td><p>Scan window size on both side of the summit
(default: 100bp)</p></td>
    </tr>
    <tr>
      <th>buffer_size</th>
      <td><p>Buffer size for incrementally increasing
internal array size to store reads alignment information. In
most cases, you don't have to change this parameter. However,
if there are large number of chromosomes/contigs/scaffolds in
your alignment, it's recommended to specify a smaller buffer
size in order to decrease memory usage (but it will take longer
time to read alignment files). Minimum memory requested for
reading an alignment file is about # of CHROMOSOME *
BUFFER_SIZE * 8 Bytes. DEFAULT: 100000</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Set verbose level. 0: only show critical message, 1:
show additional warning message, 2: show process information,
3: show debug messages. If you want to know where are the
duplicate reads, use 3. DEFAULT:2</p></td>
    </tr>
    <tr>
      <th>outdir</th>
      <td><p>The output directory.</p></td>
    </tr>
    <tr>
      <th>outputfile</th>
      <td><p>Output bedGraph file name. If not specified, will
write to standard output. REQUIRED.</p></td>
    </tr>
    <tr>
      <th>log</th>
      <td><p>Whether to capture logs.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>macsList</code> object.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>eh</span> <span class='op'>&lt;-</span> <span class='fu'>ExperimentHub</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html'>ExperimentHub</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2020-12-09</span></div><div class='input'><span class='va'>CHIP</span> <span class='op'>&lt;-</span> <span class='va'>eh</span><span class='op'>[[</span><span class='st'>"EH4558"</span><span class='op'>]</span><span class='op'>]</span>
</div><div class='output co'>#&gt; <span class='message'>see ?MACSdata and browseVignettes('MACSdata') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>CTRL</span> <span class='op'>&lt;-</span> <span class='va'>eh</span><span class='op'>[[</span><span class='st'>"EH4563"</span><span class='op'>]</span><span class='op'>]</span>
</div><div class='output co'>#&gt; <span class='message'>see ?MACSdata and browseVignettes('MACSdata') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'><span class='va'>res</span> <span class='op'>&lt;-</span> <span class='fu'><a href='callpeak.html'>callpeak</a></span><span class='op'>(</span><span class='va'>CHIP</span>, <span class='va'>CTRL</span>, gsize <span class='op'>=</span> <span class='fl'>5.2e7</span>, cutoff_analysis <span class='op'>=</span> <span class='cn'>TRUE</span>,
outdir <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>, name <span class='op'>=</span> <span class='st'>"callpeak_narrow0"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: </span>
#&gt; <span class='message'># Command line: </span>
#&gt; <span class='message'># ARGUMENTS LIST:</span>
#&gt; <span class='message'># name = callpeak_narrow0</span>
#&gt; <span class='message'># format = AUTO</span>
#&gt; <span class='message'># ChIP-seq file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c68423d6b_4601']</span>
#&gt; <span class='message'># control file = ['/Users/qi28068/Library/Caches/ExperimentHub/3f8c54ce7146_4606']</span>
#&gt; <span class='message'># effective genome size = 5.20e+07</span>
#&gt; <span class='message'># band width = 300</span>
#&gt; <span class='message'># model fold = [5.0, 50.0]</span>
#&gt; <span class='message'># qvalue cutoff = 5.00e-02</span>
#&gt; <span class='message'># The maximum gap between significant sites is assigned as the read length/tag size.</span>
#&gt; <span class='message'># The minimum length of peaks is assigned as the predicted fragment length "d".</span>
#&gt; <span class='message'># Larger dataset will be scaled towards smaller dataset.</span>
#&gt; <span class='message'># Range for calculating regional lambda is: 1000 bps and 10000 bps</span>
#&gt; <span class='message'># Broad region calling is off</span>
#&gt; <span class='message'># Additional cutoff on fold-enrichment is: 0.10</span>
#&gt; <span class='message'># Paired-End mode is off</span>
#&gt; <span class='message'> </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 read tag files... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 read treatment tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: Detected format is: BED </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: * Input file is gzipped. </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1.2 read input tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: Detected format is: BED </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: * Input file is gzipped. </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 tag size is determined as 101 bps </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 tag size = 101.0 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1  total tags in treatment: 49622 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 user defined the maximum tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1  tags after filtering in treatment: 48047 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1  Redundant rate of treatment: 0.03 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1  total tags in control: 50837 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 user defined the maximum tags... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1  tags after filtering in control: 50783 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1  Redundant rate of control: 0.00 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #1 finished! </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 Build Peak Model... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 looking for paired plus/minus strand peaks... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 Total number of paired peaks: 1036 </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 Model building with cross-correlation: Done </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 finished! </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 predicted fragment length is 230 bps </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2 alternative fragment length(s) may be 230,238 bps </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #2.2 Generate R script for model : /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_model.r </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #3 Call peaks... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #3 Pre-compute pvalue-qvalue table... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:37: #3 Cutoff vs peaks called will be analyzed! </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:38: #3 Analysis of cutoff vs num of peaks or total length has been saved in b'/var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_cutoff_analysis.txt' </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:38: #3 Call peaks for each chromosome... </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:38: #4 Write output xls file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_peaks.xls </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:38: #4 Write peak in narrowPeak format file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_peaks.narrowPeak </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:38: #4 Write summits bed file... /var/folders/pv/qm6yts0j7hdgggvq53hrzfr0vfyhvv/T//RtmpsaYg38/callpeak_narrow0_summits.bed </span>
#&gt; <span class='message'>INFO  @ Sun, 13 Dec 2020 14:23:38: Done! </span>
#&gt; <span class='message'></span></div><div class='output co'>#&gt; <span class='error'>Error in .setupMethodsTables(fdef, initialize = TRUE): trying to get slot "group" from an object of a basic class ("NULL") with no slots</span></div><div class='input'><span class='fu'>refinepeak</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='st'>"narrowPeak"</span>, <span class='va'>res</span><span class='op'>$</span><span class='va'>outputs</span>, value <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>, <span class='va'>CHIP</span>,
outdir <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>, outputfile <span class='op'>=</span> <span class='st'>"refine"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in grep("narrowPeak", res$outputs, value = TRUE): object 'res' not found</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Qiang Hu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


